internal/restconfig/restconfig.go

```diff
@@ -256,121 +320,115 @@ func (rcp *Producer) RunOnSelectedPrefixedContext(prefix string, function PerCon
-func (rcp *Producer) AddKubeContextMultiFlag(cmd *cobra.Command, usage string) {
-	rcp.AddKubeConfigFlag(cmd)
+// RunOnAllContexts runs the given function on all accessible non-prefixed contexts.
+// If the user has explicitly selected one or more contexts, only those contexts are used.
+// All appropriate contexts are processed, and any errors are aggregated.
+// Returns an error if no contexts are found.
+func (rcp *Producer) RunOnAllContexts(function PerContextFn, status reporter.Interface) error {
+	if rcp.inCluster {
+		return runInCluster(function, status)
+	}
 
-	if usage == "" {
-		usage = "comma-separated list of kubeconfig contexts to use, can be specified multiple times.\n" +
-			"If none specified, all contexts referenced by the kubeconfig are used"
+	if rcp.defaultClientConfig == nil {
+		// If we get here, no context was set up, which means SetupFlags() wasn't called
+		return status.Error(errors.New("no context provided (this is a programming error)"), "")
 	}
 
-	cmd.PersistentFlags().StringSliceVar(&rcp.kubeContexts, "kubecontexts", nil, usage)
-}
+	if rcp.defaultClientConfig.overrides.CurrentContext != "" {
+		// The user has explicitly chosen a context, use that only
+		return rcp.RunOnSelectedContext(function, status)
+	}
 
-// AddInClusterConfigFlag adds a flag enabling in-cluster configurations for processes running in pods.
-func (rcp *Producer) AddInClusterConfigFlag(cmd *cobra.Command) {
-	cmd.PersistentFlags().BoolVar(&rcp.inCluster, "in-cluster", false, "use the in-cluster configuration to connect to Kubernetes")
-}
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		rcp.defaultClientConfig.loadingRules, rcp.defaultClientConfig.overrides)
 
-func (rcp *Producer) PopulateTestFramework() {
-	framework.TestContext.KubeContexts = rcp.kubeContexts
-	if rcp.kubeConfig != "" {
-		framework.TestContext.KubeConfig = rcp.kubeConfig
+	rawConfig, err := clientConfig.RawConfig()
+	if err != nil {
+		return status.Error(err, "error retrieving the raw kubeconfig setup")
 	}
-}
 
-func (rcp *Producer) MustGetForClusters() []RestConfig {
-	configs, err := rcp.getRestConfigs()
-	exit.OnErrorWithMessage(err, "Error getting REST Config for cluster")
+	contextErrors := []error{}
+	processedContexts := 0
 
-	return configs
-}
+	if len(rcp.contexts) > 0 {
+		// Loop over explicitly-chosen contexts
+		for _, contextName := range rcp.contexts {
+			processedContexts++
 
-func (rcp *Producer) CountRequestedClusters() int {
-	if len(rcp.kubeContexts) > 0 {
-		// Count unique contexts
-		contexts := stringset.New()
-		for i := range rcp.kubeContexts {
-			contexts.Add(rcp.kubeContexts[i])
-		}
+			chosenContext, ok := rawConfig.Contexts[contextName]
+			if !ok {
+				contextErrors = append(contextErrors, fmt.Errorf("error retrieving context %s", contextName))
+				status.Failure("error retrieving context %s", contextName)
```

Comment:
```suggestion
				contextErrors = append(contextErrors, status.Error(fmt.Errorf("Kubernetes context %q not found", contextName), ""))
```

Reply:
Hah, this time the linter flags the leading capital letter!